src/components/select/option/option.component.js

```diff
@@ -0,0 +1,50 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import StyledOption from './option.style';
+
+const Option = ({
+  text,
+  children,
+  onSelect,
+  value,
+  index,
+  selectedIndex
+}) => {
+  const [isHighlighted, setIsHighlighted] = useState();
+  useEffect(() => {
+    setIsHighlighted(index === selectedIndex);
+  }, [index, selectedIndex]);
+
+  function handleClick() {
+    onSelect({ text, value });
+  }
+
+  return (
+    <StyledOption
+      aria-selected={ isHighlighted }
+      data-component='option'
+      onClick={ handleClick }
+      isHighlighted={ isHighlighted }
+      role='option'
+    >
+      { children || text }
+    </StyledOption>
+  );
+};
+
+Option.propTypes = {
+  /** The option's visible text, displayed within Textbox of Select, and used for filtering */
+  text: PropTypes.string.isRequired,
+  /** Optional: alternative rendered content, displayed within SelectList of Select (eg: an icon, an image, etc) */
+  children: PropTypes.node,
```

Comment:
Having the option of `text` and `children` causes this weird bug where you can select an option that does not appear in the list.

https://codesandbox.io/s/select-review-vbx73

Reply:
This is an expected behaviour, please look at the description above the `children` prop. We have allowed to do it in the past version of the component.

Reply:
+1 from my side on this being a weird behaviour

Reply:
Can we get confirmation from UX that they want this behaviour. Now is our chance to remove legacy defects. cc @harpalsingh

Reply:
Hello - thanks for pointing me at this thread.
I'd imagine that an <option> would always have some kind of text label (mandatory), and other optional elements (thumbnail image, icon...)
I can't think of an example where an <option> would not have a text label, and I agree that it isn't expected behaviour to appear to choose an item from the list, but then the selected value is then different.
I may be missing some context or detail coming into this fresh, but that's my suggestion anyway.
Happy to discuss if you would like?

Reply:
@benwilsonsage what do you propose to be done here?

Reply:
Discussed in detail with Kuba and considered various options.
The tricky part is backwards compatibility with existing implementations - which could have a variety of formats.
We discussed possible checks to prevent the symptom above, but can't be sure this wouldn't result in a breaking change, or prevent a valid implementation. For example, perhaps there may be an implementation that implements text and sub-text, where both option and children are required.
Ultimately it could be argued that the developer can use the flexibility wisely, or implement unexpected behaviour, but that's a factor of their design and implementation.

Given that we couldn't find a practical solution right now, I'd suggest we accept it as a legacy feature, although should be used wisely. 
Although that's my suggestion, caveat that my knowledge of code conventions and best practice is very limited!